yt/utilities/exceptions.py

```diff
@@ -20,13 +20,12 @@ def __init__(self, filename, args=None, kwargs=None):
         self.kwargs = kwargs
 
     def __str__(self):
-        msg = f"Could not determine input format from {self.filename}"
-        if self.args is not None:
-            msg += ", {self.args}"
-        if self.kwargs is not None:
-            msg += f", {self.kwargs}"
-        msg += "."
-        return msg
+        msg = [f"Could not determine input format from `'{self.filename}'"]
+        if self.args not in (None, ()):
```

Comment:
I'm not convinced this is the place we should be handling empty tuples and dicts if they're passed.

Reply:
Is it a problem of any sort if they are ?
I mean this should only ever happen because `yt.load()` passes down its `args` and `kwargs` to the error, even if they are empty... in fact I'm thinking it doesn't make much sense to default to None for both here, they should simply be defined as
```
__init__(filename, *args, **kwargs)
```
to make it consistent. *Then* we won't need to test for None at all because we know for sure that they are a tuple and a dictionnary respectively, in which case I can revert to `if self.args:` !

Reply:
refactored _again_. I like it much more now, I hope you will too :)